### PR TITLE
treewide: fix darwin build errors due to missing `IOKit` framework

### PR DESCRIPTION
**Pull request by eliandoran**
## Description of changes

Solves packages failing due to:

```
  = note: ld: framework not found IOKit
```

All modified packages should build just fine, except `rcp` which fails (and thus marked as broken) due to:

```
       >   = note: Undefined symbols for architecture x86_64:
       >             "_utimensat", referenced from:
       >                 nix::sys::stat::utimensat::hcbcd556c98a2ad7f in rcp-3d56da0b226b7fd3.common-0e282218e2519163.common.efff97c1573bd2a8-cgu.3.rcgu.o.rcgu.o
       >                 nix::with_nix_path_allocating::h8d4a1aa59714ebfc (.llvm.13116499662928166544) in rcp-3d56da0b226b7fd3.common-0e282218e2519163.common.efff97c1573bd2a8-cgu.4.rcgu.o.rcgu.o
       >           ld: symbol(s) not found for architecture x86_64
       >           clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
Are the packages missing `IOKit` or `Security`?

**Comment by eliandoran** [comment]
@wegank , it's `IOKit`. Copy-paste mistake on my side.
I will make similar PRs with other frameworks.

File: pkgs/by-name/rc/rcp/package.nix [hunk 1]
```diff
@@ -30,5 +36,7 @@ rustPlatform.buildRustPackage rec {
     license = with licenses; [ mit ];
     mainProgram = "rcp";
     maintainers = with maintainers; [ wykurz ];
+    # = note: Undefined symbols for architecture x86_64: "_utimensat"
+    broken = stdenv.isDarwin && stdenv.isx86_64;
```
Comment (wegank) [inline pkgs/by-name/rc/rcp/package.nix:40]: Perhaps using `darwin.apple_sdk_11_0.{frameworks,rustPlatform}` might help?

**Comment by eliandoran** [comment]
Result of `nixpkgs-review pr 312654` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>rcp</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>nufmt</li>
    <li>python311Packages.pystatgrab</li>
    <li>python311Packages.pystatgrab.dist</li>
    <li>python312Packages.pystatgrab</li>
    <li>python312Packages.pystatgrab.dist</li>
  </ul>
</details>

File: pkgs/by-name/rc/rcp/package.nix (hunk 1 above)
Comment (eliandoran) [reply pkgs/by-name/rc/rcp/package.nix:40]: @wegank , unfortunately that does not seem to solve it.

There are two packages marked as broken due to `_utimensat`:
- rustpython
- rubyfmt

Interestingly enough, a patch in `qemu` from some time ago says it should have been fixed in SDK 11:

```nix
# QEMU upstream does not demand compatibility to pre-10.13, so 9p-darwin
# support on nix requires utimensat fallback. The patch adding this fallback
# set was removed during the process of upstreaming this functionality, and
# will still be needed in nix until the macOS SDK reaches 10.13+.
```

Comment (wegank) [reply pkgs/by-name/rc/rcp/package.nix:40]: Oh, `rustpython` is broken again? Then `darwin.apple_sdk_11_0.rustPlatform` is probably fundamentally broken, I suppose...

Comment (eliandoran) [reply pkgs/by-name/rc/rcp/package.nix:40]: Indeed, it was marked as broken on darwin-x86_64 about 6 months ago: https://github.com/NixOS/nixpkgs/commit/b27c3af657c9cc41751fd756d1480c783532fb11

I will give `darwin.apple_sdk_11_0.rustPlatform` a try in #312508, maybe it'll work there.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 312654` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nufmt</li>
    <li>python311Packages.pystatgrab</li>
    <li>python311Packages.pystatgrab.dist</li>
    <li>python312Packages.pystatgrab</li>
    <li>python312Packages.pystatgrab.dist</li>
    <li>rcp</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Comment by eliandoran** [comment]
Result of `nixpkgs-review pr 312654` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by IogaMaster — APPROVED** [review]

**Pull request merged**